enigma/src/main/java/cuchaz/enigma/translation/mapping/serde/MappingParseException.java

```diff
@@ -12,23 +12,27 @@
 package cuchaz.enigma.translation.mapping.serde;
 
 import java.io.File;
-import java.util.function.Supplier;
 
 public class MappingParseException extends Exception {
 	private int line;
 	private String message;
 	private String filePath;
 
 	public MappingParseException(File file, int line, String message) {
+		this(file.getAbsolutePath(), line, message);
+	}
+
+	public MappingParseException(String filename, int line, String message) {
 		this.line = line;
 		this.message = message;
-		filePath = file.getAbsolutePath();
+		filePath = filename;
 	}
 
-	public MappingParseException(Supplier<String> filenameProvider, int line, String message) {
```

Comment:
Would it not be better to just take the `Path` instead?

Reply:
Good suggestion!